src/lib/u2/daos.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'DAOList',
+  extends: 'foam.u2.View',
+  documentation: 'Expects its data to be a DAO, and renders one viewFactory ' +
+      'for each one.',
+
+  topics: [ 'rowClick' ],
+
+  properties: [
+    {
+      class: 'Function',
+      name: 'rowFactory',
+      required: true,
+      adapt: function(old, nu) {
+        if ( typeof nu === 'function' ) return nu;
+
+        if ( typeof nu === 'string' ) {
+          return function(args, ctx) {
+            return ctx.lookup(nu).create(args, ctx);
+          };
+        } else {
```

Comment:
No need for an 'else' if you've already returned, but I don't mind if this is how you prefer to code.